Finmer.Game/Gameplay/SceneCombat2.cs

```diff
@@ -641,12 +662,17 @@ private void ShowUIGrappling(GameUI ui)
                 // Player is being pinned
                 ui.Instruction = $"You're being pinned down by {m_Player.GrapplingWith.Character.Name}! What will you do?";
 
-                ui.AddButton(new ChoiceButtonModel
+                // IF player cannot start the grapple against larger target, make sure they cannot reverse it either.
+                var grapple_targets = GetViableGrappleTargets(m_Player);
```

Comment:
This won't work correctly because this function gives you a list of ALL suitable grappling targets in a combat session. But right now, the player is already grappling, so that one Participant they're grappling with is the only thing that should determine whether the player can Reverse. Try calling the new CanGrapple function on the player character, passing their `GrapplingWith` partner as argument.